File: Apprentice2022b-GoogleCodeJam-main/Kotlin/d1000000.kt
fun main(args: Array<String>) {
    // input of number of cases (T)
    var cases = readln().toInt();
    // Array of results
    var results = IntArray(cases);
    // Input of Cases + storing results of each
    for (i in 1..cases){
        results[i-1] = calculateStraight();
    }
    //Printing the results
    for (i in 1..cases){
        println("Case #$i: " + results[i-1]);
    }
}

// Function to calculate max straight number
fun calculateStraight(): Int {
    // Input of number of dices
    var dices = readln().toInt();
    // Input of Array of dices (Int)
    var diceSides = readln().split(" ").map { it.toInt() }
    //We sort the array
    diceSides = diceSides.sorted()
    //We only need two parameters to know if we can keep a straight (min)
    var minSides = diceSides[0];
    var maxStraight: Int = 0;

    // For loop to calculate maxStraight
    for (sides in diceSides){
        if (sides >= minSides && sides  > maxStraight){
            maxStraight += 1;
        }else{
            continue;
        }
    }

    //return result Int
    return maxStraight;
}

File: Apprentice2022b-GoogleCodeJam-main/Python/d1000000.py
# Function to calculate max straight number
def calculateStraight():
    dices = int(input())
    # Array of dices
    diceSides = list(map(int, input().split()))
    # We sort the array for convinience O(n)
    diceSides.sort()
    # We only need two parameters to know if we can keep a straight
    minSides = diceSides[0]
    maxStraight = 0

    # For loop to calculate maxStraight
    for i in range(dices):
        # O(n) Function
        if (diceSides[i] >= minSides) and (diceSides[i] > maxStraight):
            maxStraight += 1
        else:
            continue

    # Our function returns an int
    return maxStraight


# Input routine
cases = int(input())

# Array to store results for further printing
results = []
# O(n^2) but since we know #cases then our function is O(nLogn) time
for i in range(cases):
    results.append(calculateStraight())
# Printing Routine
for i in range(cases):
    print(f"Case #{i + 1}: {results[i]}")
